DuggaSys/diagram.js

```diff
@@ -1137,19 +1149,41 @@ function generateContextProperties()
     if (context.length == 1)
     {
         var element = context[0];
-        
         //ID MUST START WITH "elementProperty_"!!!!!1111!!!!!1111 
         for (const property in element) {
             switch (property.toLowerCase()) {
                 case "name":
+
                     str += `<input id="elementProperty_${property}" type="text" value="${element[property]}" onfocus="propFieldSelected(true)" onblur="propFieldSelected(false)"> `;
                     break;
             
                 default:
                     break;
             }
         }
-        str+=`<br><br><input type="submit" value="Save" onclick="saveProperties()">`;
+        //Creates drop down for changing state of ER elements
+        if(element.kind=="ERAttr"){
+            str += '<select id="propertySelect">';
+                for (i = 0; i < attrState.length; i++) {
+                    str += '<option value='+attrState[i]+'>'+ attrState[i] +'</option>';
+                }
+            str += '</select>';
+        }
+        else if(element.kind=="EREntity"){
+            str += '<select id="propertySelect">';
+                for (i = 0; i < entityState.length; i++) {
+                    str += '<option value='+entityState[i]+'>'+ entityState[i] +'</option>';
+                }
+            str += '</select>';
+        }
+        else if(element.kind=="ERRelation"){
+            str += '<select id="propertySelect">';
+                for (i = 0; i < relationState.length; i++) {
+                    str += '<option value='+relationState[i]+'>'+ relationState[i] +'</option>';
+                }
+            str += '</select>';
+        }
+        str+=`<br><br><input type="submit" value="Save" onclick="changeState();saveProperties()">`;
```

Comment:
A lot of copy-pasted code here. Use a single for-loop and apply the value on the string depending on object type instead.

Reply:
You can store the value before performing the loop and then apply the value like so:
     
     var value;
     if (element.kind == foo)
         value = bar[i]
         
Then use:

     str += '<option value=' + value + '>' + relationState[i] +'</option>';